lib/matplotlib/ticker.py

```diff
@@ -2583,6 +2585,64 @@ def view_limits(self, vmin, vmax):
         return result
 
 
+class AsinhLocator(Locator):
+    """
+    An axis tick locator specialized for the inverse-sinh scale
+
+    This is very unlikely to have any use beyond the AsinhScale class.
+    """
+    def __init__(self, linear_width, numticks=11):
+        """
+        Parameters
+        ----------
+        linear_width : float
+            The scale parameter defining the extent
+            of the quasi-linear region.
+        numticks : int, default: 12
+            The approximate number of major ticks that will fit
+            along the entire axis
+        """
+        super().__init__()
+        self.linear_width = linear_width
+        self.numticks = numticks
+
+    def set_params(self, numticks=None):
+        """Set parameters within this locator."""
+        if numticks is not None:
+            self.numticks = numticks
+
+    def __call__(self):
+        dmin, dmax = self.axis.get_data_interval()
+        return self.tick_values(dmin, dmax)
+
+    def tick_values(self, vmin, vmax):
+        # Construct a set of "on-screen" locations
+        # that are uniformly spaced:
+        ymin, ymax = self.linear_width * np.arcsinh(np.array([vmin, vmax])
+                                                        / self.linear_width)
+        ys = np.linspace(ymin, ymax, self.numticks)
+        zero_dev = np.abs(ys / (ymax - ymin))
+        if (ymin * ymax) < 0 and min(zero_dev) > 1e-6:
+            # Ensure that the zero tick-mark is included,
+            # if the axis stradles zero
+            ys = np.hstack([ys[(zero_dev > 0.5 / self.numticks)], 0.0])
```

Comment:
Looks like you need a test for this case?

Reply:
I believe this is already covered by the `test_linear_values()` and `test_wide_values()` [methods](https://github.com/rwpenney/matplotlib/blob/feature/asinh-scale/lib/matplotlib/tests/test_ticker.py#L460), which involve axis-ranges that are symmetric around the origin. During development, some of those tests *did* fail without this check on `min(zero_dev)`. Visual inspection has also confirmed that various demonstration plots are less likely to have tick-collisions near zero.

Reply:
If we keep this Locator, I still see some tick collisions around zero (asinh demo and colormap normalization), so I wonder if this should be "widened" or set to some threshold based on linear_width to try and remove some more ticks even.